src/parser/csql_grammar.y

```diff
@@ -21791,6 +21796,11 @@ identifier
 
 			    size_in = strlen(str_name);
 
+			    if (!strchr (str_name, '.') && size_in >= (DB_MAX_IDENTIFIER_LENGTH - 1))
+			      {
+				str_name[DB_MAX_IDENTIFIER_LENGTH - 1] = '\0';	// truncate
+			      }
+
```

Comment:
I would like to comment on the same structure here and below.

It comes after performing a 286 length excess check in csql_lexer.l .
Is there any reason to check DB_MAX_IDENTIFIER_LENGTH again here?

Reply:
It's a change from a recent commit, but I'd like to explain it.

Except for user_specified_name, other identifiers are limited to 255 characters.
In order to store user_specified_name as it is, a maximum of 287 bytes is allowed in csql_lexer.l.
However, identifiers that should be limited to 255 bytes can be allowed up to 287 bytes, so check once more in csql_grammar.y.
If not user_specified_name, truncate it to 255 bytes.

During team discussion, we decided to accept the suggestion that user_specified_name should also be limited to 255 bytes.
So there is no need to allow up to 287 bytes in csql_lexer.l, and there is no need to check again in csql_grammar.y.
Because we limit user_specified_name to 255 bytes, class_name stores 223 bytes excluding 32 bytes storing user_name.